tests/framework/extensions/provisioning/verify.go

```diff
@@ -420,24 +445,52 @@ func VerifySnapshots(client *rancher.Client, localclusterID string, clusterName
 	return snapshotToBeRestored, err
 }
 
-// getSnapshots is a helper function to get the snapshots for a cluster
-func GetSnapshots(client *rancher.Client, localclusterID string, clusterName string) ([]string, error) {
+// GetRKE1Snapshots is a helper function to get the existing snapshots for a downstream RKE1 cluster.
+func GetRKE1Snapshots(client *rancher.Client, clusterName string) ([]string, error) {
+	clusterID, err := clusters.GetClusterIDByName(client, clusterName)
+	if err != nil {
+		return nil, err
+	}
+
+	snapshotSteveObjList, err := client.Management.EtcdBackup.ListAll(&types.ListOpts{
+		Filters: map[string]interface{}{
+			"clusterId": clusterID,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	snapshots := []string{}
+	for _, snapshot := range snapshotSteveObjList.Data {
+		if strings.Contains(snapshot.Name, clusterID) {
+			snapshots = append(snapshots, snapshot.ID)
+		}
+	}
+
+	return snapshots, nil
+}
+
+// GetRKE2K3SSnapshots is a helper function to get the existing snapshots for a downstream RKE2/K3S cluster.
+func GetRKE2K3SSnapshots(client *rancher.Client, localclusterID string, clusterName string) ([]string, error) {
```

Comment:
same nit for this, public vs. private

Reply:
Same as above, RBAC testing uses this test.